src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -193,7 +196,7 @@ private static bool ShouldSkipSerialization(ObjectTypeProperty property, InputMo
             return inputProperty.IsReadOnly;
         }
 
-        private void GetConstructorParameters(IEnumerable<Parameter> parameters, out List<Parameter> fullParameterList, out IEnumerable<Parameter> parametersToPassToBase, bool isInitializer, Func<Parameter, Parameter> creator)
+        private void GetConstructorParameters(IEnumerable<Parameter> parameters, out List<Parameter> fullParameterList, out IEnumerable<Parameter> parametersToPassToBase, bool isInitializer)
```

Comment:
we no longer need the `creator` because this list of parameters was taken from `ModelTypeProviderFields`'s different property.
If we are creating public ctor, we take parameters from `fields.PublicConstructorParameters` and if we are creating serialization ctor, we take parameters from `fields.SerializationParameters`.
It makes more sense that we could prepare everything in those lists in `ModelTypeProviderFields`.